quinn-h3/src/data.rs

```diff
@@ -191,7 +191,7 @@ enum WriteFrameState {
 
 impl<F> WriteFrame<F>
 where
-    F: FrameHeader + IntoPayload,
+    F: FrameHeader + Payload,
```

Comment:
What's the story here?

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

Should I rename it to `AsPayload` with an `as_payload` function?

Reply:
Renamed it.

Reply:
Ah, skimmed past the trait being renamed below. This seems fine to me as-is.